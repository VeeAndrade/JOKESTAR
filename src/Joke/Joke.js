import React from 'react';
import './Joke.scss';

const Joke = joke => {
  const { setup, punchline} = joke.joke
  return(
    <article className='joke-wrapper'>
      <h3 className='joke-setup'>{setup}</h3>
      <p className='joke-punchline'>{punchline}</p>
    </article>
  )
}

export default Joke;