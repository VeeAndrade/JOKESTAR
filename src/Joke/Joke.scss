.joke-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.joke-punchline,
.joke-setup {
  color: white;
  font-size: 1.7em;
}

.joke-punchline {
  font-size: 1.5em;
}