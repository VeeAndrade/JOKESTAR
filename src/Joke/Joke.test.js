import React from 'react';
import { Joke } from './Joke';
import { shallow } from 'enzyme';

describe('Joke', () => {

  let wrapper

  beforeEach(() => {
    wrapper = shallow( <Joke
    joke={{id: 42, setup: 'how much does a polarbear weigh?', delivery: 'enough to bresk the ice'}}
    />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
}); 