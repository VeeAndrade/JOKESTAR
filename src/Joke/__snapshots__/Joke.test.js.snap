// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joke should match the snapshot 1`] = `
<article
  className="joke-wrapper"
>
  <h3
    className="joke-setup"
  >
    how much does a polarbear weigh?
  </h3>
  <p
    className="joke-punchline"
  >
    enough to bresk the ice
  </p>
</article>
`;
